Tests/SwiftAlgorithmsTests/SuffixTests.swift

```diff
@@ -0,0 +1,20 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import Algorithms
+
+final class SuffixTests: XCTestCase {
```

Comment:
Tests should be added to handle:
- Empty input
- Empty output (predicate doesn’t match the first element)
- Predicate matches just the first element
- Predicate matches all elements